### ISSUE TITLE
Chemacs2 related: doom/sandbox fails to load vanilla Doom

### ISSUE THREAD

**Issue by Jonghyun-Yun** [doomemacs/doomemacs#5845]
### What did you expect to happen?

[Chemacs2](https://github.com/plexus/chemacs2) is occupying `~/emacs.d`, and Doom is installed in `~/doom-emacs`.  When I run `doom/sanbox`, and press `C-c C-d`, the vanilla Doom should be launched. 

The below is my `.emacs-profiles.el`:
```
(("default"   . ((user-emacs-directory . "~/doom-emacs")
                 (env . (("DOOMDIR" . "~/.doom.d")))))
 ("spacemacs" . ((user-emacs-directory . "~/spacemacs")))
 ("scimax" . ((user-emacs-directory . "~/scimax")))
```

### What actually happened?

`doom/sandbox` cannot locate `autoloads.27.2`. It looks for `~/emacs.d` instead of `~/doom-emacs`.
Here is the backtrace.

```
﻿Debugger entered--Lisp error: (file-missing "Cannot open load file" "No such file or directory" "~/.emacs.d/.local/autoloads.27.2")
  load("~/.emacs.d/.local/autoloads.27.2" nil nomessage)
  (condition-case e (load (string-remove-suffix ".el" doom-autoloads-file) nil 'nomessage) ((debug file-missing) (if (locate-file doom-autoloads-file load-path) (signal 'doom-autoload-error e) (signal 'doom-error (list "Doom is in an incomplete state" "run 'doom sync' on the command line to repair it")))))
  (progn (setq doom-init-p t) (if doom-debug-p (progn (let ((inhibit-message (active-minibuffer-window))) (message #("DOOM Initializing Doom" 0 5 (face font-lock-comment-face)))))) (let ((--dolist-tail-- '(exec-path load-path))) (while --dolist-tail-- (let ((var (car --dolist-tail--))) (set-default var (get var 'initial-value)) (setq --dolist-tail-- (cdr --dolist-tail--))))) (condition-case e (load (string-remove-suffix ".el" doom-autoloads-file) nil 'nomessage) ((debug file-missing) (if (locate-file doom-autoloads-file load-path) (signal 'doom-autoload-error e) (signal 'doom-error (list "Doom is in an incomplete state" "run 'doom sync' on the command line to repair it"))))) (if doom-debug-p (doom-debug-mode 1)) (if (and (or (display-graphic-p) (daemonp)) doom-env-file) (progn (progn (set-default 'process-environment (get 'process-environment 'initial-value))) (doom-load-envvars-file doom-env-file 'noerror))) (require 'core-modules) (autoload 'doom-initialize-packages "core-packages") (eval-after-load 'package #'(lambda nil (require 'core-packages))) (eval-after-load 'straight #'(lambda nil (doom-initialize-packages))) (if noninteractive nil (add-hook 'after-change-major-mode-hook #'doom-run-local-var-hooks-maybe-h 100) (add-hook 'hack-local-variables-hook #'doom-run-local-var-hooks-h) (add-hook 'emacs-startup-hook #'doom-load-packages-incrementally-h) (add-hook 'window-setup-hook #'doom-display-benchmark-h) (doom-run-hook-on 'doom-first-buffer-hook '(find-file-hook doom-switch-buffer-hook)) (doom-run-hook-on 'doom-first-file-hook '(find-file-hook dired-initial-position-hook)) (doom-run-hook-on 'doom-first-input-hook '(pre-command-hook)) (add-hook 'doom-first-buffer-hook #'gcmh-mode)))
  (if (or force-p (not doom-init-p)) (progn (setq doom-init-p t) (if doom-debug-p (progn (let ((inhibit-message (active-minibuffer-window))) (message #("DOOM Initializing Doom" 0 5 (face font-lock-comment-face)))))) (let ((--dolist-tail-- '(exec-path load-path))) (while --dolist-tail-- (let ((var (car --dolist-tail--))) (set-default var (get var 'initial-value)) (setq --dolist-tail-- (cdr --dolist-tail--))))) (condition-case e (load (string-remove-suffix ".el" doom-autoloads-file) nil 'nomessage) ((debug file-missing) (if (locate-file doom-autoloads-file load-path) (signal 'doom-autoload-error e) (signal 'doom-error (list "Doom is in an incomplete state" "run 'doom sync' on the command line to repair it"))))) (if doom-debug-p (doom-debug-mode 1)) (if (and (or (display-graphic-p) (daemonp)) doom-env-file) (progn (progn (set-default 'process-environment (get 'process-environment 'initial-value))) (doom-load-envvars-file doom-env-file 'noerror))) (require 'core-modules) (autoload 'doom-initialize-packages "core-packages") (eval-after-load 'package #'(lambda nil (require 'core-packages))) (eval-after-load 'straight #'(lambda nil (doom-initialize-packages))) (if noninteractive nil (add-hook 'after-change-major-mode-hook #'doom-run-local-var-hooks-maybe-h 100) (add-hook 'hack-local-variables-hook #'doom-run-local-var-hooks-h) (add-hook 'emacs-startup-hook #'doom-load-packages-incrementally-h) (add-hook 'window-setup-hook #'doom-display-benchmark-h) (doom-run-hook-on 'doom-first-buffer-hook '(find-file-hook doom-switch-buffer-hook)) (doom-run-hook-on 'doom-first-file-hook '(find-file-hook dired-initial-position-hook)) (doom-run-hook-on 'doom-first-input-hook '(pre-command-hook)) (add-hook 'doom-first-buffer-hook #'gcmh-mode))))
  doom-initialize()
  (let ((doom-init-modules-p t)) (doom-initialize) (doom-initialize-core-modules))
  eval-buffer(#<buffer  *load*> nil "/var/folders/47/35cm9bpx23nbn9q369yc31v40000gq/T/d..." nil t)  ; Reading at buffer position 29711
  load-with-code-conversion("/var/folders/47/35cm9bpx23nbn9q369yc31v40000gq/T/d..." "/var/folders/47/35cm9bpx23nbn9q369yc31v40000gq/T/d..." nil t)
  load("/var/folders/47/35cm9bpx23nbn9q369yc31v40000gq/T/d..." nil t)
  command-line-1(("-l" "/var/folders/47/35cm9bpx23nbn9q369yc31v40000gq/T/d..."))
  command-line()
  normal-top-level()
```

### Describe your attempts to resolve the issue

Symlinking `~/doom-emacs` to `~/emacs.d` after launching the Doom profile makes `doom/sandbox` work as expected.

### Steps to reproduce

A fresh installation of Doom with no user configuration results in the same error.

### System Information

https://pastebin.com/r9Mk4Aip

**Comment by hlissner** [comment]
I am aware of this issue, but is currently low priority (and _may_ be addressed as part of  #4273). I'll welcome a PR to fix it ahead of me.

**Comment by hlissner** [comment]
Sorry for the delay. As of https://github.com/doomemacs/doomemacs/commit/60083a2626dc5f5e91c39d85b65259d838e06a69 and https://github.com/doomemacs/doomemacs/commit/a022e55c08a18cfed97c0695da45553f7f73ab82, `doom/sandbox` should be fixed. Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!
